src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,207 +1,850 @@
-            else # Cone
-                (T(rbotMin)..T(rbotMax), T(rtopMin)..T(rtopMax))
-            end
-        elseif rMin_is_zero #Cone with rMin = 0
-            (T(rbotMax), T(rtopMax))
-        else # Cone
-            (T(rbotMin)..T(rbotMax), T(rtopMin)..T(rtopMax))
-        end
-    φ = mod(T(φMax) - T(φMin), T(2π)) == 0 ? nothing : T(φMin)..T(φMax)
-    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
-    Cone( T, r, φ, z)
+function _in(pt::CartesianPoint, c::PartialCylinder{T,ClosedPrimitive}) where {T} 
+    az = abs(pt.z) 
+    (az <= c.hZ || csg_isapprox_lin(az, c.hZ)) && begin
+        r = hypot(pt.x, pt.y) 
+        r <= c.r || csg_isapprox_lin(r, c.r)
+    end &&
+    _in_angular_interval_closed(atan(pt.y, pt.x), c.φ)
+end
+function _in(pt::CartesianPoint, c::PartialCylinder{T,OpenPrimitive}) where {T} 
+    abs(pt.z) < c.hZ &&
+    hypot(pt.x, pt.y) < c.r &&
+    _in_angular_interval_open(atan(pt.y, pt.x), c.φ)
 end
-Cone(rbotMin, rbotMax, rtopMin, rtopMax, φMin, φMax, zMin, zMax) = Cone(; rbotMin = rbotMin, rbotMax = rbotMax, rtopMin = rtopMin, rtopMax = rtopMax, φMin = φMin, φMax = φMax, zMin = zMin, zMax = zMax)
 
-function Cone(rbot::R1, rtop::R2, height::H) where {R1<:Real, R2<:Real, H<:Real}
-    T = float(promote_type(R1, R2, H))
-    Cone( T, (T(rbot), T(rtop)), nothing, T(height)/2)
+function surfaces(t::PartialCylinder{T}) where {T}
+    bot_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), -t.hZ), t) 
+    top_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), +t.hZ), t) 
+    mantle = PartialConeMantle{T,:inwards}((t.r, t.r), t.φ, t.hZ, t.origin, t.rotation)
+    e_bot = PartialCircularArea{T}(r = t.r, φ = t.φ, origin = bot_center_pt, rotation = t.rotation)
+    e_top = PartialCircularArea{T}(r = t.r, φ = t.φ, origin = top_center_pt, rotation = -t.rotation * RotZ{T}(π))
+    poly_l = _transform_into_global_coordinate_system(Quadrangle{T}((
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[1], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], -t.hZ)) )), t)
+    poly_r = _transform_into_global_coordinate_system(Quadrangle{T}((
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], +t.hZ)) )), t)
+    # normals of the surfaces show inside the volume primitives. 
+    (e_top, e_bot, mantle, poly_l, poly_r)
 end
 
+####################################################################
+####################################################################
+
+#=
+___
+   \
+    \
+_____\
+
+=#
+const VaryingCylinder{T,CO} = Cone{T,CO,Tuple{T,T},Nothing} # Full in φ
```

Comment:
`Varying` as in varying radius over `z`? I find this name a bit odd..

Reply:
Yes, that is the idea. 
Fell free to come up with something more elegant. 
It's easy to replace.